RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc

```diff
@@ -294,30 +291,17 @@ void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
     constexpr double meanoffset  = meanlimlow + 0.5*(meanlimhigh-meanlimlow);
     constexpr double meanscale   = 0.5*(meanlimhigh-meanlimlow);
     
-    constexpr double sigmalimlow  = 0.0002;
-    constexpr double sigmalimhigh = 0.5;
-    constexpr double sigmaoffset  = sigmalimlow + 0.5*(sigmalimhigh-sigmalimlow);
-    constexpr double sigmascale   = 0.5*(sigmalimhigh-sigmalimlow);  
-    
     const GBRForestD *forestmean = iseb ? foresteb_ : forestee_;
-    const GBRForestD *forestsigma = iseb ? forestsigmaeb_ : forestsigmaee_;
-    
-    //these are the actual BDT responses
+
     double rawmean = forestmean->GetResponse(eval.data());
-    double rawsigma = forestsigma->GetResponse(eval.data());
-    
-    //apply transformation to limited output range (matching the training)
     double mean = meanoffset + meanscale*vdt::fast_sin(rawmean);
-    double sigma = sigmaoffset + sigmascale*vdt::fast_sin(rawsigma);
 
     double ecor = mean*eval[6];
     if (!iseb)  
-      ecor = mean*(eval[6]+sc.preshowerEnergy());
-    const double sigmacor = sigma*ecor;
+      ecor = mean*eval[6]+sc.preshowerEnergy();
    
     sc.setEnergy(ecor);
     sc.setCorrectedEnergy(ecor);
-    sc.setCorrectedEnergyUncertainty(sigmacor);
```

Comment:
Hi Slava,
at HLT we are not using it so I was trying to reduce as much as possible
the unnecessary operations for the timing.
I can make a couple of measurement to see the impact of having it or not.

Regards,
     Matteo.

+++++++++++
Matteo Sani
University of California, San Diego, US.
Tel: +41227671577
CERN Office: 40-3-A02


On Thu, Feb 18, 2016 at 6:55 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc
> <https://github.com/cms-sw/cmssw/pull/13333#discussion_r53353076>:
>
> >
> >      sc.setEnergy(ecor);
> >      sc.setCorrectedEnergy(ecor);
> > -    sc.setCorrectedEnergyUncertainty(sigmacor);
>
> is sigmacor really not needed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13333/files#r53353076>.
>